R/data.table.R

```diff
@@ -611,7 +612,7 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
           if (identical(nomatch, 0L) && allLen1) irows = irows[irows != 0L]
         } else {
           if (length(xo) && missing(on))
-            stop("Cannot by=.EACHI when joining to a secondary key, yet")
+            stop("Internal error. Cannot by=.EACHI when joining to a secondary key, yet") # nocov
```

Comment:
Unless I'm missing something, it's not possible to hit this branch. Thrown to `nocov` since it's hinting at some future functionality? I might be missing something...